katharsis-client/src/main/java/io/katharsis/client/KatharsisClient.java

```diff
@@ -0,0 +1,226 @@
+	}
+
+	private void initRepositories() {
+		// register all resources
+		ResourceLookup resourceLookup = moduleRegistry.getResourceLookup();
+		Set<Class<?>> resourceClasses = resourceLookup.getResourceClasses();
+		for (Class<?> resourceClass : resourceClasses) {
+			allocateRepository(resourceClass);
+		}
+	}
+
+	private void initExceptionMapperRegistry() {
+		ExceptionMapperLookup exceptionMapperLookup = moduleRegistry.getExceptionMapperLookup();
+		exceptionMapperRegistry = new ExceptionMapperRegistryBuilder().build(exceptionMapperLookup);
+	}
+
+	@SuppressWarnings({ "rawtypes", "unchecked" })
+	private <T, ID extends Serializable> void allocateRepository(Class<T> resourceClass) {
+		ResourceInformation resourceInformation = moduleRegistry.getResourceInformationBuilder().build(resourceClass);
+		final ResourceRepositoryStub<T, ID> repositoryStub = new ResourceRepositoryStubImpl<T, ID>(this, resourceClass,
+				resourceInformation, urlBuilder);
+
+		// create interface for it!
+		RepositoryInstanceBuilder repositoryInstanceBuilder = new RepositoryInstanceBuilder(null, null) {
+			@Override
+			public Object buildRepository() {
+				return repositoryStub;
+			}
+		};
+		ResourceEntry<T, ID> resourceEntry = new DirectResponseResourceEntry<T, ID>(repositoryInstanceBuilder);
+		Set<ResourceField> relationshipFields = resourceInformation.getRelationshipFields();
+		List<ResponseRelationshipEntry<T, ?>> relationshipEntries = new ArrayList<ResponseRelationshipEntry<T, ?>>();
+		RegistryEntry<T> registryEntry = new RegistryEntry<T>(resourceInformation, resourceEntry, relationshipEntries);
+
+		for (ResourceField relationshipField : relationshipFields) {
+			final Class<?> targetClass = relationshipField.getType();
+			final RelationshipRepositoryStubImpl relationshipRepositoryStub = new RelationshipRepositoryStubImpl(this,
+					resourceClass, targetClass, resourceInformation, urlBuilder, registryEntry);
+			RepositoryInstanceBuilder<RelationshipRepository> relationshipRepositoryInstanceBuilder = new RepositoryInstanceBuilder<RelationshipRepository>(
+					null, null) {
+				@Override
+				public RelationshipRepository buildRepository() {
+					return relationshipRepositoryStub;
+				}
+			};
+			DirectResponseRelationshipEntry relationshipEntry = new DirectResponseRelationshipEntry(
+					relationshipRepositoryInstanceBuilder) {
+
+				@Override
+				public Class<?> getTargetAffiliation() {
+					return targetClass;
+				}
+			};
+			relationshipEntries.add(relationshipEntry);
+		}
+		resourceRegistry.addEntry(resourceClass, registryEntry);
+	}
+
+	/**
+	 * @param resourceClass
+	 * @return stub for the given resourceClass
+	 */
+	@SuppressWarnings({ "unchecked", "rawtypes" })
+	public <T, ID extends Serializable> ResourceRepositoryStub<T, ID> getRepository(Class<T> resourceClass) {
+		init();
+
+		RegistryEntry<T> entry = resourceRegistry.getEntry(resourceClass);
+
+		// TODO fix this in katharsis, should be able to get original resource
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1135)